ui/tests/e2e/fixtures/environmentsPage.js

```diff
@@ -0,0 +1,221 @@
+import { expect } from '@playwright/test';
+import { ENV } from '../env';
+
+export class EnvironmentPage {
+  /**
+   * @param {import('@playwright/test').Page} page
+   */
+  constructor(page) {
+    this.page = page;
+  }
+  async navigate() {
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+  }
+  async fillInput(locator, value) {
+    await locator.fill(value);
+  }
+
+  async goToProfiles() {
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+  }
+
+  async goToConnections() {
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/connections`);
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/connections`);
+  }
+
+  async doesProfileExist(environmentName) {
+    await this.goToProfiles();
+    return this.page
+      .locator(`text=${environmentName}`)
+      .count()
+      .then((count) => count === 1);
+  }
+  async createEnvironmentProfile(environmentName) {
+    if (!(await this.doesProfileExist(environmentName))) {
+      await this.page.getByRole('button', { name: 'Create' }).click();
+      await this.fillInput(this.page.locator('#Environment_name'), `${environmentName}`);
+      await this.fillInput(
+        this.page.locator('#Environment_description'),
+        'Created profile for playwright tests',
+      );
+      await this.page.getByRole('button', { name: 'Save', exact: true }).click();
+      // Check for notification visibility
+      const notification = await this.page.locator(`text=Environment "${environmentName}" created`);
+      await expect(notification).toBeVisible();
+      await this.page.waitForTimeout(8000);
```

Comment:
curious why using waitForTimeOut. It will add some flakiness.

https://playwright.dev/docs/api/class-page#page-wait-for-timeout

Reply:
I abhor timeouts. Sometimes they aren't avoidable. I'm hoping that this isn't one of those times, though.

Reply:
@yash37158 This approach makes the test slower, use toPass instead

https://playwright.dev/docs/test-assertions#expecttopass

Reply:
The timeout happens because it takes a bit of time to successfully generate the Environment card after receiving the notification. I've adjusted my approach for this